hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/grantapproval/ClassicGrantApprovalCall.java

```diff
@@ -76,7 +77,10 @@ public PricedResult execute() {
                     : GrantApprovalTranslator.GRANT_APPROVAL_NFT.getOutputs().encodeElements((long)
                             status.protoOrdinal());
 
-            return gasOnly(FullResult.successResult(encodedOutput, gasRequirement), status, false);
+            return gasOnly(
+                    FullResult.successResult(encodedOutput, gasRequirement, (ContractCallRecordBuilder) recordBuilder),
```

Comment:
Instead of this cast it seems nicer to update [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/grantapproval/ClassicGrantApprovalCall.java#L63) to dispatch with a type of `ContractCallRecordBuilder.class`. 🤔

Reply:
Thanks.